File: PepesComing/Sprites.vb
Imports Microsoft.Xna.Framework
Imports Microsoft.Xna.Framework.Graphics
Imports Microsoft.Xna.Framework.Input

Public Class Sprites
    Private tileset As Texture2D

    Public Shared ReadOnly Property SandWall As Rectangle
        Get
            Dim rndint As Integer = Game.rnd.Next(1, 100)
            If rndint < 80 Then
                Return NormalSandWall
            ElseIf rndint < 90 Then
                Return DamagedSandWall1
            Else
                Return DamagedSandWall2
            End If
        End Get
    End Property

    Public Shared ReadOnly NormalSandWall As Rectangle = New Rectangle(4 * 17, 20 * 17, 16, 16)
    Public Shared ReadOnly DamagedSandWall1 As Rectangle = New Rectangle(4 * 17, 21 * 17, 16, 16)
    Public Shared ReadOnly DamagedSandWall2 As Rectangle = New Rectangle(3 * 17, 21 * 17, 16, 16)
    Public Shared ReadOnly Start As Rectangle = New Rectangle(9 * 17, 24 * 17, 16, 16)
    Public Shared ReadOnly EndPoint As Rectangle = New Rectangle(11 * 17, 24 * 17, 16, 16)
    Public Shared ReadOnly Property SteelFloor As Rectangle = New Rectangle(23 * 17, 1 * 17, 16, 16)


    Sub New(game As Game)
        tileset = game.Content.Load(Of Texture2D)("Tileset.png")
    End Sub

    Sub DrawTile(spritebatch As SpriteBatch, destination As Rectangle, tile As Rectangle)
        spritebatch.Draw(tileset, destination, tile, Color.White)
    End Sub
End Class

File: PepesComing.ConvertedToC#/Sprites.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Linq;
using System.Xml.Linq;
using System.Threading.Tasks;
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;
using Microsoft.Xna.Framework.Input;
namespace PepesComing {

    public class Sprites {
        public Texture2D Texture { get; private set; }

        public static Rectangle SandWall {
            get {
                int rndint = Game.rnd.Next(1, 100);
                if (rndint < 80) {
                    return NormalSandWall;
                } else if (rndint < 90) {
                    return DamagedSandWall1;
                } else {
                    return DamagedSandWall2;
                }
            }
        }

        // Walls
        public static readonly Rectangle NormalSandWall = new Rectangle(4 * 17, 20 * 17, 16, 16);
        public static readonly Rectangle DamagedSandWall1 = new Rectangle(4 * 17, 21 * 17, 16, 16);
        public static readonly Rectangle DamagedSandWall2 = new Rectangle(3 * 17, 21 * 17, 16, 16);

        // Start/end
        public static readonly Rectangle Start = new Rectangle(9 * 17, 24 * 17, 16, 16);
        public static readonly Rectangle EndPoint = new Rectangle(11 * 17, 24 * 17, 16, 16);

        // Floors
        public static readonly Rectangle SteelFloor = new Rectangle(23 * 17, 1 * 17, 16, 16);

        public Sprites(Game game) {
            Texture = game.Content.Load<Texture2D>("Tileset.png");
        }
    }
}
